core/cache/machine.go

```diff
@@ -174,12 +174,13 @@ func (m *Machine) setDetails(details MachineChange) {
 
 	m.setStale(false)
 
-	if details.InstanceId != m.details.InstanceId {
+	provisioned := details.InstanceId != m.details.InstanceId
+	m.details = details
```

Comment:
This makes sense, we end up publishing an event before we've set the details, which could in turn cause people to read an empty details part. Which would cause confusion in the very least.